experimental/traffic-portal/src/app/core/statuses/status-details/status-details.component.ts

```diff
@@ -0,0 +1,175 @@
+			description: new FormControl("",Validators.required),
+			name: new FormControl("",Validators.required),
+		});
+
+		// Getting id from the route
+		this.id = this.route.snapshot.paramMap.get("id");
+
+		// we check whether params is a number if not we shall assume user wants to add a new status.
+		if (!this.isNew) {
+			this.loading = true;
+			this.statusDetailsForm.addControl("id", new FormControl(""));
+			this.statusDetailsForm.addControl("lastUpdated", new FormControl(""));
+			this.getStatusDetails();
+		} else {
+			this.navSvc.headerTitle.next("New Status");
+		}
+	}
+
+	/**
+	 * Reloads the servers table data.
+	 *
+	 * @param id is the id passed in route for this page if this is a edit view.
+	 */
+	public async getStatusDetails(): Promise<void> {
+		const id = Number(this.id) ; // id Type 'null' is not assignable to type 'string'
+		this.statusDetails = await this.serverService.getStatuses(id);
+		const data: ResponseStatus = {
+			description: this.statusDetails.description,
+			id: this.statusDetails.id,
+			lastUpdated: new Date(),
+			name: this.statusDetails.name
+		};
+
+		// Set page title with status ID
+		this.navSvc.headerTitle.next(`Status #${data.id}`);
+
+		// Patch the form with existing data we got from service requested above.
+		this.statusDetailsForm.patchValue(data);
+		this.loading = false;
+	}
+
+	/**
+	 * On submitting the form we check for whether we are performing Create or Edit
+	 */
+	public onSubmit(): void {
+		if (this.isNew) {
+			this.createStatus();
+
+		} else {
+			this.updateStatus();
+		}
+	}
+
+	/**
+	 * For Creating a new status
+	 */
+	public createStatus(): void {
+		this.serverService.createStatus(this.statusDetailsForm.value).then((res: ResponseStatus) => {
+			if (res) {
+				this.id = (res?.id);
+				this.router.navigate([`/core/statuses/${this.id}`]);
+				this.navSvc.headerTitle.next(`Status #${this.id}`);
+			}
+		});
+	}
+
+	/**
+	 * For updating the Status
+	 */
+	public updateStatus(): void {
+		this.serverService.updateStatusDetail(this.statusDetailsForm.value, Number(this.id));
+	}
+
+	/**
+	 * Deleteting status
+	 */
+	public async deleteStatus(): Promise<void> {
+		const ref = this.dialog.open<DecisionDialogComponent, DecisionDialogData, boolean>(DecisionDialogComponent, {
+			data: {
+				message: `This action CANNOT be undone. This will permanently delete '${this.statusDetails?.name}'.`,
+				title: `Delete Status: ${this.statusDetails?.name}`
+			}
+		});
+
+		ref.afterClosed().subscribe(result => {
+			if (result) {
+				const id = Number(this.id);
+				this.serverService.deleteStatus(id).then(() => {
+					this.router.navigate(["/core/statuses"]);
+				});
+			}
+		});
+	}
+
+	/**
+	 * Title for the page
+	 */
+	public get title(): string {
+		return this.isNew ? "Add New Status" : "Edit Status";
+	}
+
+	/**
+	 * Checking for params to ensure given id is a number
+	 */
+	public get isNew(): boolean {
+		return this.id === "new" && isNaN(Number(this.id));
```

Comment:
the second part of this condition is redundant, because `isNaN(Number("new"))` is a static value: `true`.